File: common/src/main/java/com/pluxity/permission/PermissionGroupRepository.java
package com.pluxity.permission;

import org.springframework.data.jpa.repository.JpaRepository;

public interface PermissionGroupRepository extends JpaRepository<PermissionGroup, Long> {
    boolean existsByName(String permissionGroupName);

    boolean existsByNameAndIdNot(String newName, Long id);
}

File: common/src/main/java/com/pluxity/permission/PermissionGroupRepository.kt
package com.pluxity.permission

import org.springframework.data.jpa.repository.JpaRepository

interface PermissionGroupRepository : JpaRepository<PermissionGroup, Long> {
    fun existsByName(permissionGroupName: String): Boolean

    fun existsByNameAndIdNot(
        newName: String,
        id: Long,
    ): Boolean
}
